### DISCUSSION TITLE
pre-commit hook

### DISCUSSION THREAD

**Discussion by Niccolum** [npm/feedback#529] [Registry]
As we know, we have a lot of pre-commit hooks tools. I use pre-commit (https://pre-commit.com) and it will be great, if npm will support it (write .pre-commit-hooks.yaml config), for example, for run npm audit as hook before commit. Can you support it?

**Comment by ljharb** [comment]
You can already do that by invoking npm audit in your git hook - what would you imagine npm would do that’s different?

Note that it’s very very dangerous for tools to automatically write stuff to git hooks; there’s a reason why you can’t use git to commit an activated git hook.

**Reply by Niccolum** [reply]
This tool require special config for including it in pre-commit tool. Only that stops me) so, my question is - is it possible to include this config in npm repo?
This tool helps to use different hooks in one hook and easy to support zoo of hooks)

**Reply by ljharb** [reply]
What special config would be required for using any tool in a commit hook, with pre-commit or without?

**Reply by Niccolum** [reply]
> What special config would be required for using any tool in a commit hook, with pre-commit or without?

examples:
https://github.com/pre-commit/pre-commit-hooks/blob/master/.pre-commit-hooks.yaml
https://github.com/pre-commit/mirrors-eslint/blob/master/.pre-commit-hooks.yaml

**Reply by ljharb** [reply]
That looks like pre-commit itself needs to be configured so it knows how to use npm - nothing would need to happen in npm itself for that.

**Reply by Niccolum** [reply]
hmm. I trying and write about it here. Thanks!
